src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -638,9 +633,54 @@ public void addRtpExtension(RtpExtension rtpExtension)
     }
 
     @Override
-    public void setVideoConstraints(Map<String, VideoConstraints> newVideoConstraints)
+    public void setSenderVideoConstraints(ImmutableMap<String, VideoConstraints> newVideoConstraints)
     {
+        ImmutableMap<String, VideoConstraints>
+            oldVideoConstraints = bitrateController.getVideoConstraints();
+
         bitrateController.setVideoConstraints(newVideoConstraints);
+
+        Set<String> endpointsThatNoLongerCare
+            = new HashSet<>(oldVideoConstraints.keySet());
+        endpointsThatNoLongerCare.removeAll(newVideoConstraints.keySet());
+
+        // Endpoints that no longer care what they receive.
+        for (String id : endpointsThatNoLongerCare)
+        {
+            AbstractEndpoint senderEndpoint = getConference().getEndpoint(id);
+            if (senderEndpoint != null)
+            {
+                senderEndpoint.removeReceiver(getID());
+            }
+        }
+
+        // Added or updated.
+        for (Map.Entry<String, VideoConstraints>
+            videoConstraintsEntry : newVideoConstraints.entrySet())
+        {
+            AbstractEndpoint senderEndpoint
+                = getConference().getEndpoint(videoConstraintsEntry.getKey());
+
+            if (senderEndpoint != null)
+            {
+                senderEndpoint.addReceiver(
+                    getID(), videoConstraintsEntry.getValue());
+            }
+        }
+    }
+
+    @Override
+    protected void onMaxReceiverVideoConstraintsChanged(VideoConstraints maxVideoConstraints)
+    {
+        // Note that it's up to the client to respect these constraints.
+        String senderVideoConstraintsMessage = createSenderVideoConstraintsMessage(maxVideoConstraints);
+
+        if (logger.isDebugEnabled())
+        {
+            logger.debug("Is now selected, sending message: " + senderVideoConstraintsMessage);
```

Comment:
This log message seems inaccurate, it's not necessarily selected, right?  Just constraints changed?